packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -0,0 +1,227 @@
+        removed = true;
+      }
+    }
+    if (removed) {
+      this.dispatchContextsState();
+    }
+  }
+
+  createKubeContextInformers(context: KubeContext): ContextInternalState | undefined {
+    const kc = new KubeConfig();
+    const cluster = this.kubeConfig.clusters.find(c => c.name === context.cluster);
+    if (!cluster) {
+      return;
+    }
+    const user = this.kubeConfig.users.find(u => u.name === context.user);
+    if (!user) {
+      return;
+    }
+    kc.loadFromOptions({
+      clusters: [cluster],
+      users: [user],
+      contexts: [context],
+      currentContext: context.name,
+    });
+
+    const ns = context.namespace ?? 'default';
+    return {
+      podInformer: this.createPodInformer(kc, ns, context),
+      deploymentInformer: this.createDeploymentInformer(kc, ns, context),
+    };
+  }
+
+  private createPodInformer(kc: KubeConfig, ns: string, context: KubeContext): Informer<V1Pod> & ObjectCache<V1Pod> {
+    const k8sApi = kc.makeApiClient(CoreV1Api);
+    const listFn = () => k8sApi.listNamespacedPod(ns);
+    const path = `/api/v1/namespaces/${ns}/pods`;
+    return this.createInformer<V1Pod>(kc, context, path, listFn, {
+      onAdd: _obj => this.safeSetState(context.name, state => state.podsCount++),
+      onDelete: _obj => this.safeSetState(context.name, state => state.podsCount--),
+      onReachable: reachable => this.safeSetState(context.name, state => (state.reachable = reachable)),
+    });
+  }
+
+  private createDeploymentInformer(
+    kc: KubeConfig,
+    ns: string,
+    context: KubeContext,
+  ): Informer<V1Deployment> & ObjectCache<V1Deployment> {
+    const k8sApi = kc.makeApiClient(AppsV1Api);
+    const listFn = () => k8sApi.listNamespacedDeployment(ns);
+    const path = `/apis/apps/v1/namespaces/${ns}/deployments`;
+    return this.createInformer<V1Deployment>(kc, context, path, listFn, {
+      onAdd: _obj => this.safeSetState(context.name, state => state.deploymentsCount++),
+      onDelete: _obj => this.safeSetState(context.name, state => state.deploymentsCount--),
+    });
+  }
+
+  private createInformer<T extends KubernetesObject>(
+    kc: KubeConfig,
+    context: KubeContext,
+    path: string,
+    listPromiseFn: ListPromise<T>,
+    options: CreateInformerOptions<T>,
+  ): Informer<T> & ObjectCache<T> {
+    const informer = makeInformer(kc, path, listPromiseFn);
+
+    informer.on('add', (obj: T) => {
+      options.onAdd?.(obj);
+      options.onReachable?.(true);
+    });
+
+    informer.on('delete', (obj: T) => {
+      options.onDelete?.(obj);
+      options.onReachable?.(true);
+    });
+    informer.on('error', (err: unknown) => {
+      if (err !== undefined) {
+        console.error(`informer error on path ${path} for context ${context.name}: `, String(err));
+      }
+      options.onReachable?.(err === undefined);
+      // Restart informer after 5sec
+      setTimeout(() => {
+        this.restartInformer<T>(informer, context, options);
+      }, 5000);
```

Comment:
where is kept that setTimeout reference ?

if there is like several errors, would it add two times restartInformer ?

Reply:
I have added a commit with timer related changes